ecs-agent/tmds/handlers/taskprotection/v1/handlers/handlers.go

```diff
@@ -59,6 +61,10 @@ type TaskProtectionRequest struct {
 	ExpiresInMinutes  *int64
 }
 
+type CanceledError interface {
```

Comment:
please add a comment explaining what this is useful for, and a link to https://pkg.go.dev/github.com/aws/smithy-go#CanceledError.CanceledError